### PR TITLE
launcher: add fix for long socket paths

### LINKED ISSUE #124: Long console socket path.
Integration tests of start command do not work correctly on Mac OS because the console socket paths are too long. Max path length is 108 on Linux, 104 on Mac OS. When running integration tests on my machine path length exceeds 140, instances fail, however tests pass.

### PR DESCRIPTION
**Pull request by better0fdead**
Now relative path is used instead of real for console socket.
It is used to prevent console socket path from being too long.
Closes #124

### CODE REVIEW

File: cli/connector/connector.go [hunk 1]
```diff
@@ -37,6 +39,17 @@ type Connector interface {
 
 // Connect connects to the tarantool instance according to options.
 func Connect(opts ConnectOpts) (Connector, error) {
+	// It became common that address is longer than 140 symbols(limit).
```
Comment (0x501D) [inline cli/connector/connector.go:42]: On Linux sun_path is 108 bytes length.
Anyway we should check the minimal length and throw an error if  opts.Address > 108.

Comment (better0fdead) [reply cli/connector/connector.go:42]: That patch allows us to use opts.Address longer than sun_path limits.

Comment (better0fdead) [reply cli/connector/connector.go]: Changed comment

Comment (0x501D) [reply cli/connector/connector.go]: Suppose this: 
```go
opts.Address = "/tmp/aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
opts.Address = "./" + filepath.Base(opts.Address)
```
pts.Address length will be > 108

Comment (better0fdead) [reply cli/connector/connector.go]: Changed comments prior to your message

**Comment by 0x501D** [comment]
Please, rebase with master to resolve conflicts.

**Comment by better0fdead** [comment]
> Please, rebase with master to resolve conflicts.

Done

**Comment by better0fdead** [comment]
blocked by [issue](https://github.com/tarantool/cartridge-cli/issues/736)

File: cli/connector/connector.go [hunk 2]
```diff
@@ -37,6 +39,23 @@ type Connector interface {
...
+	// e.g foo/bar/123.sock -> ./123.sock
+	workDir, err := os.Getwd()
+	if err != nil {
+		return nil, err
+	}
+	if _, err := os.Stat(opts.Address); err == nil {
+		os.Chdir(filepath.Dir(opts.Address))
+		opts.Address = "./" + filepath.Base(opts.Address)
+		if len(opts.Address) > 108 {
+			return nil, fmt.Errorf("socket name is longer than 106 symbols: %s",
```
Comment (0x501D) [inline cli/connector/connector.go:54]: 106->108

File: cli/running/instance.go [hunk 3]
```diff
@@ -104,9 +104,26 @@ func (inst *Instance) Start() error {
...
-	inst.Cmd.Env = append(inst.Cmd.Env,
-		"TT_CLI_CONSOLE_SOCKET="+inst.consoleSocket)
 
+	// It became common that console socket path is longer than 108 symbols(sun_path limit).
+	// To reduce length of path we use relative path
+	// with chdir into a directory of console socket.
+	// e.g foo/bar/123.sock -> ./123.sock
+	if inst.consoleSocket != "" {
+		if len("./"+filepath.Base(inst.consoleSocket)) > 108 {
+			return fmt.Errorf("socket name is longer than 106 symbols: %s",
```
Comment (0x501D) [inline cli/running/instance.go:114]: 106->108, Use constant var.

Comment (Totktonada) [reply cli/running/instance.go:114]: 108 on Linux, 104 on Mac OS AFAIR. But that's with leading `'\0'`. So 107 and 103.

Comment (Totktonada) [reply cli/running/instance.go:114]: See also a remark regarding using 108th and 104th byte in https://github.com/tarantool/tarantool/issues/4634 and links regarding Mac OS.

Comment (0x501D) [reply cli/running/instance.go:114]: from unix(7):
The length of the pathname, including the terminating null byte, should not exceed the size of sun_path.

File: cli/connector/connector.go (hunk 2 above)
Comment (better0fdead) [reply cli/connector/connector.go]: Done

File: cli/running/instance.go (hunk 3 above)
Comment (better0fdead) [reply cli/running/instance.go]: Done

**Review by 0x501D — APPROVED** [review]
LGTM

**Review by LeonidVas — COMMENTED** [review]
Hi! Thank you for the patchset.
I don't (and will not) review the code, and I think a review from Pavel is sufficient in this case. Please edit the commit messages and I will merge PR.

"cartridge: update to 2.12.3" -> "cartridge-cli: bump 2.12.3 version"
- it would be nice to describe the reason for updating the submodule in the commit message

"CI: add explicit license for config package" -> "ci: ..."
https://github.com/tarantool/tarantool/wiki/Code-review-procedure#commit-message

> ... started from lowercase letter ...
> Commit message should fit 72 symbols line width ...

**Comment by LeonidVas** [comment]
Add a description of the changes to the CHANGELOG.

**Pull request merged**